### ISSUE TITLE
Upgrade to a supported version of Django

### ISSUE THREAD

**Issue by karindalziel** [open-oni/open-oni#8]
Duplicate issue from chronam: https://github.com/LibraryOfCongress/chronam/issues/118

According to https://www.djangoproject.com/download/ the current supported versions of Django are 1.4 LTS (until October 2015), 1.7 (December 2015) and 1.8 LTS (April 2018).

 Replace South with the core migration feature added in Django 1.7

**Comment by karindalziel** [comment]
Glenn Bunton will take this

**Comment by mark-cooper** [comment]
I took a swipe at this today:

https://github.com/LibraryOfCongress/chronam/compare/master...open-oni:django-1.8.3?expand=1

Includes changes from LOC, a WSGI api update based on a startup error, and a fix for deprecated "mimetype" (requires "content_type"). Also requires:

https://github.com/open-oni/docker-open-oni/compare/django-1.8.3?expand=1

To test locally 1) clone everything 2) checkout django-1.8.3 branches 3) run the dev script (or run mysql, solr, build open-oni-dev and run).

Result: the website appears =)

Loaded batch. Can view images after troubleshooting (all changes are in those branches). Don't see any obvious issues ... a good first pass I think.

**Comment by jduss4** [comment]
I can verify that 1.8.3 works.  I only tested it on a few pages, newspapers.json, newspapers.rdf, and a bit of random clicking around.  However, I had trouble getting the new open-oni docker image to work with the `dev.sh` script until I changed 
```
   echo "Starting chronam for development ... "
   // etc
   -v $(pwd)/data:/opt/chronam/data \
    open-oni:dev
```
to 
```
    // etc
    openoni/open-oni:latest
```

It's quite possible that I am using docker incorrectly, but I did get your two branches of open-oni / docker running with django 1.8.3 once I changed that line.

**Comment by mark-cooper** [comment]
Great, thanks for testing. I really need to use Docker Machine to experience what everyone else is dealing with =)

I'm actually a kinda surprised that `openoni/open-oni:latest` would work. That's built with the old 1.5 stuff in it, so what running with that image plus those mounts would do is mix a 1.5 build with some django 1.7+ ish looking updates ... which is interesting.

What's the problem with `open-oni:dev`? You should get that image by running `docker build -t open-oni:dev -f Dockerfile-dev .` in the docker-open-oni folder. You should then be able to see it using `docker images`. You may want to try upping the DELAY in the dev script, that works fine for me on my desktop, but may be too low for a laptop running docker on a VM (this would solve the problem of mysql and / or solr not being ready during open-oni startup).

**Comment by jduss4** [comment]
Now I'm not on my home machine so I can't check for the exact errors, but when I was trying to pull the openoni/open-oni:dev image it kept saying image not found.  With `docker images` I could see open-oni but its tag was "latest" instead of "dev" which is why I ultimately said "yolo" and tried to run that instead.

It's possible that I pulled the image incorrectly, however.  I didn't run `docker build -t open-oni:dev -f Dockerfile-dev .` so perhaps that's my problem.  I was running variations on `docker pull openoni`.

Perhaps I should start everything up again and have a debug line put out to the screen the django version once the site is running to verify that it was using 1.8.3.  When the container was starting up I checked the logging to make sure it was pulling in that version of django and it looked good to go.
